### PR TITLE
Evaluates Tiles of DataSources that are not attached to a MapView as NOT visible

### PR DESCRIPTION
**Pull request by FraukeF**
Thank you for contributing to harp.gl!

Before requesting a pull request, please remember to check the following documents:
* [contribution guidelines](https://github.com/heremaps/harp.gl/blob/master/CONTRIBUTING.md)
* [coding style](https://github.com/heremaps/harp.gl/blob/master/CODINGSTYLE.md)

If you are adding new functionality we would highly appreciate if you can describe what is the capability you are adding and even better if you can add some examples. Please also remember to add tests for it.

# CI Check

Our bots will check whether your PR can be directly integrated into the mainline. We have some internal integration tests running on the background, our bots will inform you of the next steps and someone from our team will take a look and help if needed!

And please do not forget to sign-off your commit! You can read more about DCO [here](https://julien.ponge.org/blog/developer-certificate-of-origin-versus-contributor-license-agreements/). But, in short, you just need to use `git commit -s` or append `--signoff` when you are committing to the repo.

Happy contributing!

### CODE REVIEW

File: @here/harp-mapview/lib/DataSource.ts [hunk 1]
```diff
@@ -329,7 +325,10 @@ export abstract class DataSource extends THREE.EventDispatcher {
      * to a {@link MapView}.
      */
     get projection(): Projection {
-        return this.mapView.projection;
+        if (this.m_mapView === undefined) {
+            throw new Error("This DataSource was not added to MapView");
+        }
+        return (this.mapView as MapView).projection;
```
Comment (nzjony) [inline @here/harp-mapview/lib/DataSource.ts:331]:
```suggestion
        return this.m_mapView.projection;
```

File: @here/harp-mapview/lib/Tile.ts [hunk 2]
```diff
@@ -367,11 +367,17 @@ export class Tile implements CachedResource {
...
+        // this.dataSource.mapView will throw an error
+        return (
+            this.mapView !== undefined && this.frameNumLastRequested >= this.mapView.frameNumber - 1
+        );
     }
 
     set isVisible(visible: boolean) {
-        this.frameNumLastRequested = visible ? this.dataSource.mapView.frameNumber : -1;
+        this.frameNumLastRequested =
+            visible && this.mapView !== undefined ? this.mapView?.frameNumber : -1;
```
Comment (nzjony) [inline @here/harp-mapview/lib/Tile.ts:380]:
```suggestion
            visible && this.mapView !== undefined ? this.mapView.frameNumber : -1;
```

File: @here/harp-mapview/lib/DataSource.ts (hunk 1 above)
Comment (nzjony) [reply @here/harp-mapview/lib/DataSource.ts:331]: because you check for undefined above, you can use Typescripts smarts to avoid the cast

Comment (FraukeF) [reply @here/harp-mapview/lib/DataSource.ts:331]: right, nice.. will do that

File: @here/harp-mapview/lib/Tile.ts (hunk 2 above)
Comment (FraukeF) [reply @here/harp-mapview/lib/Tile.ts:380]: ok

**Comment by FraukeF** [comment]
@nzjony  I reverted to the original version as the changing of the DataSource API would cause a lot of issues deep through the code, I created HARP-11190 for a more detailed look into this.

**Pull request merged**